frontend/class-breadcrumbs.php

```diff
@@ -664,10 +664,19 @@ private function prepare_links() {
 	 * @return array Array of link text and url
 	 */
 	private function get_link_info_for_id( $id ) {
-		$link = array();
 
+		$link = array();
 		$link['url']  = get_permalink( $id );
+		$status = get_post_status( $id );
+
+		if ( 'private' == $status && ! current_user_can( 'read_private_pages' ) && ! current_user_can( 'read_private_posts' ) ) {
```

Comment:
Regarding the post status check, should the other potential statuses be addressed as well or is that overkill?

Another way would be to check if the post/page/cpt is "published" and if the current user can read it, then link it. Otherwise it does nothing. crumb_to_link() does check if 'url' is even set, so it would be valid to leave it unassigned. Would any other post status need to be linked?